### PR TITLE
feat: Introduce flag for enabling anyOf/OneOf generation

### PR DESCRIPTION
**Pull request by KavithaSiva**
## Context
 

SAP/cloud-sdk-java-backlog#264.

The latest OpenSource OpenAPI generator generates compilable clients if the input spec contains `anyOf/OneOf` keywords.

Based on a team [discussion](https://github.com/SAP/cloud-sdk-java-backlog/discussions/407), we decided to have a flag that toggles between generation/non-generation of clients for specs containing `oneOf/anyOf`.

### Feature scope:

 
- [x] Add input toggle for client generation with oneOf/anyOf
- [x] Add tests  

## Definition of Done

- [x] Functionality scope stated & covered
- [x] Tests cover the scope above
- [x] ~Error handling created / updated & covered by the tests above~
- [ ] Documentation updated
- [x] Release notes updated

### CODE REVIEW

File: datamodel/openapi/openapi-generator/src/main/java/com/sap/cloud/sdk/datamodel/openapi/generator/PreprocessingStepOrchestrator.java [hunk 1]
```diff
@@ -37,21 +38,38 @@ class PreprocessingStepOrchestrator
...
+                .warn(
+                    "oneOf/anyOf keywords processing is turned on, the generated client may not be feature complete and work as expected for all cases involving anyOf/oneOf");
+        }
+        return this;
+    }
+
     @Nonnull
     OpenApiSpec performPreprocessingSteps()
     {
+
```
Comment (Johannes-Schneider) [inline datamodel/openapi/openapi-generator/src/main/java/com/sap/cloud/sdk/datamodel/openapi/generator/PreprocessingStepOrchestrator.java:72]:
```suggestion
```

**Review by Johannes-Schneider — COMMENTED** [review]
Generally LGTM - just some minor code formatting comments.

**Review by MattKuhr — COMMENTED** [review]
Let's also add a dedicated message to the exception we throw by default so people know this option exists

File: release_notes.md [hunk 2]
```diff
@@ -20,6 +20,19 @@
 - Improve how certificates are evaluated for the authentication types `SAMLAssertion` and `OAUTH2_SAML_BEARER_ASSERTION`.
   The certificates are no longer unnecessarily parsed when obtaining the destination.
 - The `spring-boot3` archetype no longer contains the `integration-tests` module. The contained tests are now part of the test suite of the `application` module.
+- OpenAPI clients can now be generated with specifications with `oneOf` and `anyOf` keywords by enabling their processing by using `<enableOneofAnyofGeneration>` in `openapi-generator-maven-plugin`(This option is turned off by default):
+```xml
```
Comment (newtork) [inline release_notes.md:24]: <details><summary>Incorrect spacing. Please prepend two spaces to not break the list indentation.</summary>

 image

</details>

**Review by newtork — APPROVED** [review]

**Pull request merged**